src/app/common/TagList.vue

```diff
@@ -46,8 +49,9 @@
   return labels.map((tag) => {
     const { label, value } = tag
     const route = getRoute(tag)
+    const isKuma = label.includes('kuma.io/')
```

Comment:
I suppose this is vaguely valid 🤷 , `not-kuma.io` would get highlighted and we don't want that to happen, not a biggie but is there a different approach to avoid this?

Reply:
Changing this to something more robust like `label.includes('.kuma.io/') || label.startsWith('kuma.io/')` doesn’t really make things better. We can’t parse tag keys as URLs because they aren’t.

For the sake of argument we could fake the URL syntactically and use something like this:

```suggestion
    let isKuma = false
    try {
      const tagKeyAsUrl = new URL('https://' + label)
      isKuma = tagKeyAsUrl.hostname.endsWith('.kuma.io') || tagKeyAsUrl.hostname === 'kuma.io'
    } catch {}
```

That’s the most robust thing I can think of, but I also think it’s really doubtful that this is actually useful to get 100% accurate. We don’t process the result of this call other than by applying different colors.

Reply:
Well I was just thinking of fixing it instead of ignoring this alert instead, up to you. I would like to avoid this coming up in PRs either way. Would this thing appear in the Security tab in GH as well as PRs if we don't do either?

Reply:
> Well I was just thinking of fixing it instead of ignoring this alert instead, up to you. I would like to avoid this coming up in PRs either way.

What exactly needs fixing here? We’re not actually parsing URLs here so the CodeQL alerts isn’t valid. At most we’re talking about false positive matches for this `isKuma` check. `label.includes('.kuma.io/') || label.startsWith('kuma.io/')` would avoid matching `not-kuma.io`.

> Would this thing appear in the Security tab in GH as well as PRs if we don't do either?

They’ll be listed here for closed alerts: https://github.com/kumahq/kuma-gui/security/code-scanning?query=is%3Aclosed+branch%3Amaster

Reply:
Nothing _needs_ fixing, just trying to suggest something that might work and avoid all this, as I said completely up to you.

---

What about using:

```html
data-key="kuma.io/service"
```

and then targetting that with css 

```css
[data-key*=".kuma.io/"], [data-key^="kuma.io/"]
```

I would guess that would remove the alert and is probably simpler?

You could even split by `/` and use `[0]` - `data-key="kuma.io"` `[class$=".kuma.io"], [class="kuma.io"]`, or something like that?

Would any of those work?

Just to be clear, no problem to leave as is if this alert doesn't continue to alert us anywhere.

Reply:
I did change this to use `label.includes('.kuma.io/') || label.startsWith('kuma.io/')` in the off chance someone uses a non-conforming label including something liek `not-kuma.io/`, but I don’t think more caution is warranted here. The alert can be closed and won’t bother us again unless we change the code.